hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/SnapshotManager.java

```diff
@@ -877,9 +943,14 @@ public long restoreOrCloneSnapshot(final SnapshotDescription reqSnapshot, final
    */
   private long restoreSnapshot(final SnapshotDescription reqSnapshot, final TableName tableName,
       final SnapshotDescription snapshot, final TableDescriptor snapshotTableDesc,
-      final NonceKey nonceKey, final boolean restoreAcl) throws IOException {
+      final NonceKey nonceKey, final boolean restoreAcl)
+    throws IOException {
     MasterCoprocessorHost cpHost = master.getMasterCoprocessorHost();
 
+    //have to check first if restoring the snapshot would break current SFT setup
+    checkSFTCompatibility(master.getTableDescriptors().get(tableName), snapshotTableDesc,
```

Comment:
Do we really need all these complex combinations? Could we just check if the SFT conf between current descriptor and snapshot descriptor differs, then fail it?

Reply:
I think we need these things. As we can specify SFT on column family...

Reply:
That would not be fully correct. In certain scenarios we would fail even when the restore is possible and would force the user to make config changes that would not have any effect other than satisfying this check's requirements.